### DISCUSSION TITLE
Reduce unnecessary requests to backup object storage

### DISCUSSION THREAD

**Discussion by ariep** [vmware-tanzu/velero#6281] [Community support Q&A]
I'd like to avoid the very frequent requests velero makes to our object store, I assume as part of its backup sync routine. We were seeing three requests every minute, and a lot more (> 15) once every five minutes.

I changed the backupSyncPeriod of the BackupStorageLocation to `0s`. After that, the number of requests every minute decreased from 3 to 1. I'd like to get it down to 0, and also get rid of the requests every 5 minutes, but I don't know how. I also changed the server-wide backupSyncPeriod setting to `0s`, but the logs indicate that that is ignored: "Backup sync period is 1m0s".

Any ideas?

**Comment by sseago** [comment]
I'm guessing that 0 is currently being treated as "not specified, use the default of 1 minute". Not syncing at all is really not a good idea, but maybe set it to something much longer. once an hour or every few hours vs. once a minute? I'm less clear on the every-5-minutes, unless that's also relating to BSL validation. Once you get the normal sync happening less often, you can see whether these 15-every-5-minutes are still happening, and if so, what pod logs correspond with these times.

**Reply by ariep** [reply]
Hi, thanks for your input! I tried setting the sync interval for both the BackupStorageLocation and the server to one hour, but that increases the requests per minute again to 3.

**Comment by ariep** [comment]
This is in the velero logs, probably related:
```
time="2023-05-23T08:19:48Z" level=info msg="Validating BackupStorageLocation" backup-storage-location=velero/default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:131"
time="2023-05-23T08:19:48Z" level=info msg="BackupStorageLocations is valid, marking as available" backup-storage-location=velero/default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:116"
time="2023-05-23T08:20:48Z" level=info msg="Validating BackupStorageLocation" backup-storage-location=velero/default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:131"
time="2023-05-23T08:20:48Z" level=info msg="BackupStorageLocations is valid, marking as available" backup-storage-location=velero/default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:116"
time="2023-05-23T08:21:48Z" level=info msg="Validating BackupStorageLocation" backup-storage-location=velero/default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:131"
time="2023-05-23T08:21:48Z" level=info msg="BackupStorageLocations is valid, marking as available" backup-storage-location=velero/default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:116"
```
This is with `backupSyncPeriod: 1h`, so it should not be the sync.

**Reply by ariep** [reply]
Indeed, by setting the `validationFrequency` to `1h`, I could get rid of one more call every minute. So to have not calls every minute:
* set `validationFrequency` to something high
* set `backupSyncPeriod` to `0s` -- weirdly, setting this to a high number is not enough.

**Comment by ariep** [comment]
It appears that the (high number of) calls that happen every five minutes cannot be reduced:
https://github.com/vmware-tanzu/velero/issues/2820

Maybe I'll try kopia instead of restic at some point to see if that helps with this.

**Comment by sseago** [comment]
Note that if `backupSyncPeriod` is set to 0, we never sync backups. This is fine if you only ever use this BSL with the current cluster, but if velero never syncs backups, then backups in the BSL that are not (yet) available in this cluster will never be made available.
